### PR TITLE
Fix unhandled exception when viz is run with undefined parameters

### PR DESCRIPTION
**Pull request by jmholzer**
## Description
Resolves [#1228](https://github.com/kedro-org/kedro-viz/issues/1228)

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ Please follow the established format:
 ## Bug fixes and other changes
 
 - Visual fixes for metrics plots in experiment tracking. (#1229)
+- Fix bug causing user to be unable to start Kedro-Viz if undefined parameters are used. (#1231)
```
Comment (AntonyMilneQB) [inline RELEASE.md:14]:
```suggestion
- Fix bug that prevented Kedro-Viz from loading when a project contains undefined parameters. (#1231)
```

File: package/tests/test_models/test_flowchart.py [hunk 2]
```diff
@@ -228,6 +229,20 @@ def test_create_non_existing_parameter_node(self, patched_warning):
             [call("Cannot find parameter `%s` in the catalog.", "non_existing")]
         )
 
+    @patch("logging.Logger.warning")
```
Comment (AntonyMilneQB) [inline package/tests/test_models/test_flowchart.py:231]: Should we also test the `AttributeError` case (presumably caused by `self.kedro_obj = None`)? Just a thought - up to you.

**Review by AntonyMilneQB — APPROVED** [review]
Great job tracking this down and fixing it! LGTM but it would be great if you could add a bit of documentation either to this PR or as comments just so we have your current understanding of how things work recorded somewhere since it wasn't at all clear to me looking through the code. e.g. it seems that `CatalogRepository.get_dataset` also handles parameters (based purely on the fact that this fixes the bug), can `self.kedro_obj` still be `None`, etc.

On second thoughts, maybe a slightly clearer fix would be to handle the case of parameters and undefined datasets differently in `CatalogRepository.get_dataset` rather than lumping them together as `MemoryDataSet`?
```
        except DataSetNotFoundError:
            if is_parameter(dataset_name): # function is somewhere in kedro-viz, but maybe doesn't make sense to use it here depending on where it's defined...
                dataset_obj = None   
            else:
                 dataset_obj = MemoryDataSet()
```

But then on third thoughts, that would mean that `self.kedro_obj.load` never executes in `parameter_values` which doesn't sound right 🤔

Basically I think this is all good, but would be great to just explain things a bit more somewhere now that you've untangled it since there's clearly scope for confusion here!

File: RELEASE.md (hunk 1 above)
Comment (tynandebold) [reply RELEASE.md:14]: `contains` to `contained` to be past-tense consistent?

**Review by tynandebold — APPROVED** [review]
Brilliant, thank you!

**Review by rashidakanchwala — APPROVED** [review]
LGTM :)

File: package/tests/test_models/test_flowchart.py (hunk 2 above)
Comment (jmholzer) [reply package/tests/test_models/test_flowchart.py:232]: Good thought, this case is covered by the test directly above 🙂. I added comments to both tests clarifying their purpose in [786d23e](https://github.com/kedro-org/kedro-viz/pull/1231/commits/786d23e0f1cb3ad2d442a56153e619a7097d9e3e).

**Comment by jmholzer** [comment]
Thanks for the quick review and for all your help and guidance with this issue.

> Basically I think this is all good, but would be great to just explain things a bit more somewhere now that you've untangled it since there's clearly scope for confusion here!

I added an explanatory comment to the except clause in `ParametersNode.parameters_value` 🙂.

File: RELEASE.md (hunk 1 above)
Comment (jmholzer) [reply RELEASE.md]: Changed in [cf96f46](https://github.com/kedro-org/kedro-viz/pull/1231/commits/cf96f463092a0279b45e1e56d247576937416d1f).
